### PR TITLE
refactor: cleanup e2e framework job utils

### PR DESCRIPTION
**Pull request by draveness**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Refactor and clean up e2e framework job utils 

Related to #76206 

**Special notes for your reviewer**:

A function named `newBool` which placed in the job_util file has many duplications as below:

```go
test/e2e/framework/jobs_util.go:266:func newBool(val bool) *bool {
pkg/apis/batch/fuzzer/fuzzer.go:26:func newBool(val bool) *bool {
pkg/apis/batch/v1beta1/defaults_test.go:113:func newBool(val bool) *bool {
pkg/apis/batch/v2alpha1/defaults_test.go:100:func newBool(val bool) *bool {
pkg/apis/batch/validation/validation_test.go:670:func newBool(val bool) *bool {
pkg/registry/batch/job/storage/storage_test.go:181:func newBool(val bool) *bool {
pkg/registry/batch/job/strategy_test.go:36:func newBool(a bool) *bool {
```

The function returns a pointer to bool value, do you have any suggestions on where to put the function.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by jiatongw — COMMENTED** [review]
I grep the usage of this function, it shows as the follows:

```
.//pkg/apis/batch/fuzzer/fuzzer.go:    j.ManualSelector = newBool(true)
.//pkg/apis/batch/v1beta1/defaults_test.go:	Suspend:    newBool(false),
.//pkg/apis/batch/v1beta1/defaults_test.go:	Suspend:    newBool(true),
.//pkg/apis/batch/v1beta1/defaults_test.go:	Suspend:    newBool(true),
.//pkg/apis/batch/v2alpha1/defaults_test.go:	Suspend:    newBool(false),
.//pkg/apis/batch/v2alpha1/defaults_test.go:    Suspend:    newBool(true),
.//pkg/apis/batch/v2alpha1/defaults_test.go:	    Suspend:    newBool(true),
.//pkg/apis/batch/validation/validation_test.go:    ManualSelector:   newBool(true),
.//pkg/apis/batch/validation/validation_test.go:    ManualSelector:   newBool(true),
.//pkg/apis/batch/validation/validation_test.go:    ManualSelector:   newBool(true),
.//pkg/apis/batch/validation/validation_test.go:    ManualSelector:   newBool(true),
.//pkg/apis/batch/validation/validation_test.go:    ManualSelector:   newBool(true),
.//pkg/registry/batch/job/storage/storage_test.go:    ManualSelector:   newBool(true),
.//pkg/registry/batch/job/strategy_test.go:    ManualSelector:   newBool(true),
```

it seems that `pkg/apis/batch/` and `pkg/registry/batch/job/` are calling this function.  maybe we can follow the history order to put it in where it first defined?

**Review by jiatongw — COMMENTED** [review]
And those usages are just duplicate the function in the file and call it somewhere else in the same file. So we can also just put it in a more generic place and point out that people need to follow the new reference. waiting for more ideas.

**Review by neolit123 — COMMENTED** [review]
```
pkg/apis/batch/fuzzer/fuzzer.go:26:func newBool(val bool) *bool {
pkg/apis/batch/v1beta1/defaults_test.go:113:func newBool(val bool) *bool {
pkg/apis/batch/v2alpha1/defaults_test.go:100:func newBool(val bool) *bool {
pkg/apis/batch/validation/validation_test.go:670:func newBool(val bool) *bool {
pkg/registry/batch/job/storage/storage_test.go:181:func newBool(val bool) *bool {
pkg/registry/batch/job/strategy_test.go:36:func newBool(a bool) *bool {
```
please note that these packages are out of scope for the e2e testing refactor.

we should only focus on `test/`.

File: test/e2e/framework/jobs_util.go [hunk 1]
```diff
@@ -269,10 +269,8 @@ func newBool(val bool) *bool {
 	return p
 }
 
-type updateJobFunc func(*batch.Job)
```
Comment (andrewsykim) [inline test/e2e/framework/jobs_util.go]: Seems like we can do more than just this for jobs_utils :). What are your thoughts to moving the code here to a dedicated package `test/e2e/framework/jobs`?

Comment (draveness) [reply test/e2e/framework/jobs_util.go]: That sounds like a good idea. We could split the original large file into smaller one. I'll work on to it to see how it goes. :)

File: test/e2e/framework/util/job/make.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
+package job
```
Comment (andrewsykim) [inline test/e2e/framework/util/job/make.go]: I think `resource.go` would be a better name for this file, what do you think?

File: test/e2e/framework/util/job/clientset.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
+package job
```
Comment (andrewsykim) [inline test/e2e/framework/util/job/clientset.go]: Thoughts on `rest.go`?

File: test/e2e/framework/util/job/util.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+func newBool(val bool) *bool {
+	p := new(bool)
+	*p = val
+	return p
+}
+
+// FinishTime returns finish time of the specified job.
+func FinishTime(finishedJob *batch.Job) metav1.Time {
```
Comment (andrewsykim) [inline test/e2e/framework/util/job/util.go]: I don't see this used anywhere, can we remove?

File: test/e2e/framework/util/job/util.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package job
+
+import (
+	batch "k8s.io/api/batch/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func newBool(val bool) *bool {
```
Comment (andrewsykim) [inline test/e2e/framework/util/job/util.go]: Seems like we can just move this to the same file as `NewTestJob` and just remove this file entirely

File: test/e2e/framework/util/job/make.go (hunk 2 above)
Comment (andrewsykim) [reply test/e2e/framework/util/job/make.go]: or `fixtures.go`?

**Review by andrewsykim — COMMENTED** [review]
I'm not a fan of `test/e2e/framework/util/job` and prefer `test/e2e/framework/job` instead, but it's not a strongly held opinion. Structure looks good for the most part though.

File: test/e2e/framework/util/job/make.go (hunk 2 above)
Comment (draveness) [reply test/e2e/framework/util/job/make.go]: I think fixtures is a better one. :)

File: test/e2e/framework/util/job/util.go (hunk 4 above)
Comment (draveness) [reply test/e2e/framework/util/job/util.go]: The FinishTime is originaly called `JobFinishTime`, the golint warns me like this:

```go
func name will be used as job.JobFinishTime by other package, and that stutters; consider callling this FinishTime
```

File: test/e2e/framework/util/job/util.go (hunk 5 above)
Comment (draveness) [reply test/e2e/framework/util/job/util.go]: I found out that the newBool was only called in newTestJob, so I manually initialize a bool value and remove the newBool function from the pkg.

**Comment by draveness** [comment]
Do you have any idea on the name of `wait.go` file, the functions listed in the file have some overlapped usage with `rest.go`, they both read/write data with clientset.Interface.

**Comment by andrewsykim** [comment]
> Do you have any idea on the name of wait.go file, the functions listed in the file have some overlapped usage with rest.go, they both read/write data with clientset.Interface.

Either keeping them separate or combining sound good to me

**Comment by draveness** [comment]
> > Do you have any idea on the name of wait.go file, the functions listed in the file have some overlapped usage with rest.go, they both read/write data with clientset.Interface.
> 
> Either keeping them separate or combining sound good to me

@andrewsykim OK, I'll continue to work on this and fix the tests.

**Comment by draveness** [comment]
@andrewsykim @neolit123 PTAL. ;)

/hold cancel

**Comment by andrewsykim** [comment]
Thanks for iterating @draveness 

/lgtm
/assign @timothysc

**Review by timothysc — APPROVED** [review]
/approve 

I think this is fine, but I want to inspect the callsites too.  If you look through a bunch of the utility functions they only have 1 call site.  That being said, just factoring out these utilities is generally useful.

**Comment by draveness** [comment]
/test pull-kubernetes-kubemark-e2e-gce-big
--

**Comment by draveness** [comment]
/test pull-kubernetes-kubemark-e2e-gce-big
--

**Comment by draveness** [comment]
@andrewsykim @neolit123 conflicted with another PR, just rebased. Please take a look. :)

**Comment by andrewsykim** [comment]
/priority important-longterm
/lgtm

thank you @draveness

**Pull request merged**